dotCMS/src/main/java/com/dotmarketing/filters/HttpHeaderSecurityFilterWrapper.java

```diff
@@ -0,0 +1,63 @@
+package com.dotmarketing.filters;
+
+import com.dotcms.repackage.com.google.common.base.CaseFormat;
+import java.io.IOException;
+import java.util.Enumeration;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import org.apache.catalina.filters.HttpHeaderSecurityFilter;
+
+public class HttpHeaderSecurityFilterWrapper implements Filter {
+    private final HttpHeaderSecurityFilter wrappedFilter = new HttpHeaderSecurityFilter();
+
+    private static final String CONFIG_PREFIX="CMS_";
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        // Initialize the wrapped filter with custom configuration
+        FilterConfig customFilterConfig = new FilterConfig() {
+            @Override
+            public String getFilterName() {
+                return filterConfig.getFilterName();
+            }
+
+            @Override
+            public ServletContext getServletContext() {
+                return filterConfig.getServletContext();
+            }
+
+            @Override
+            public String getInitParameter(String name) {
+                // e.g. hstsEnabled is checked as environment "CMS_HSTS_ENABLED"
+                String paramValue = System.getenv(CONFIG_PREFIX+CaseFormat.LOWER_CAMEL.to(CaseFormat.UPPER_UNDERSCORE,name));
```

Comment:
Nice, it would be good to have this idea in more generic way for all web components